cityofaustin/atd-moped
# Spike - Use data grid for funding table

## Associated issues

https://github.com/cityofaustin/atd-data-tech/issues/17681

I'm finding it difficult to figure out where the stopping point for this spike is, the basic CRUD operations are here but I keep seeing things that I think need to be addressed before this is truly ready to go. 

* Inconsistent font sizes
* layout can be jumpy when going from view to edit
* you can toggle more than one row to edit at the same time
* code organization, the autocompletes are in this same component

In other words this is not ready to merge to main, but its ready for review to see how Data Grid works with row editing. 

## Testing
**URL to test:** 

https://deploy-preview-1362--atd-moped-main.netlify.app/moped/projects

**Steps to test:**
1. Test editing an existing funding record (you can edit by clicking the pencil or by double clicking in any editable cell)
2. Clicking away from the cell automatically saves your edits. You can also click the check mark to save. 
3. Test adding a blank row. Cancel without saving (clicking the X). or click away to save. 
4. Test adding a new record to a project that has an ecapris subproject funding id
5. test deleting a row

**Questions**
I pinned the actions/edit column to the left. Do we like it?
Do we want the columns or filters menu to show up? See the work activity table below. 



docs on creating own edit components
https://mui.com/x/react-data-grid/editing/#create-your-own-edit-component

---
#### Ship list
